systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java

```diff
@@ -98,11 +100,19 @@ void testKafkaClusterDowngradeToOlderMessageFormat(ExtensionContext testContext)
             initLogMsgFormat = tmp.toString();
         }
 
+        String interBrokerProtocol = sortedVersions.get(0).messageVersion();
+
+        if (interBrokerProtocol.charAt(2) + 1 >= sortedVersions.get(sortedVersions.size() - 1).messageVersion().charAt(2)) {
+            StringBuilder tmp = new StringBuilder(interBrokerProtocol);
+            tmp.setCharAt(2, (char) (interBrokerProtocol.charAt(2) - 1));
+            interBrokerProtocol = tmp.toString();
+        }
```

Comment:
This looks a bit fragile. Maybe we should parse it as dotted decimals and not assume that the 3rd character is the whole minor number.

Reply:
I agree. But it is really just copy paste of existing code. So I think we should leave that for separate PR.

Reply:
Copy and paste and no factor out common method? :scream: But I'm fine with a follow-up PR if you prefer.